helpers/focus.js

```diff
@@ -179,3 +179,12 @@ export function isFocusable(node, includeHidden, includeTabbablesOnly, includeDi
 	return _isFocusable;
 
 }
+
+export function tryApplyFocus(elem) {
+	const focusable = findComposedAncestor(elem, (node) => (isFocusable(node) || getFirstFocusableDescendant(node) !== null));
```

Comment:
In the case of dropdowns, the open button is not an ancestor of the menu items. So going up the tree, the button would never be found to focus on. Adding the check to see if any descendants are focusable allows the button to be focused on.

Reply:
If `elem` is itself focusable, wouldn't this go up the ancestor tree first and potentially find a focusable sibling? Just wondering if we should first check that `elem` itself isn't focusable before resorting to an ancestor search.

Reply:
Right.  The actual opener that gets the focus is not a direct ancestor eh?  The `getFirstFocusableDescendant` I think isn't exactly cheap and is being called recursively, but it's not the end of the work as long at `tryApplyFocus` is only being called once.

Reply:
tryApplyFocus is only called once, correct. Just when we close the dialog to find its opener to focus on. I double-checked that buttons should be caught by checking the element itself, so getFirstFocusableDescendant won't be run in those cases.